test/HexFloat.cpp

```diff
@@ -528,5 +528,442 @@ INSTANTIATE_TEST_CASE_P(
 
         })));
 
+int32_t unbiased_exponent(double f) {
+  return spvutils::HexFloat<spvutils::FloatProxy<float>>(
+      static_cast<float>(f)).getUnbiasedNormalizedExponent();
+}
+
+TEST(HexFloatOperationTest, UnbiasedExponent) {
```

Comment:
I wonder if it should be a template function over <float>, then you can test the float16 case?